### ISSUE TITLE
"system" behaves differently since dune 3 (Windows only)

### ISSUE THREAD

**Issue by fdopen** [ocaml/dune#5523]
Some packages that use "system" could be compiled with dune version 2.9.3 and earlier, but no longer with dune 3.

e.g:
mesh-easymesh.0.9.5:
https://github.com/Chris00/mesh/blob/be7f043df3dedca862ad650538f8483855836c34/easymesh/dune#L20

## Reproduction
```bash
opam install dune.3.0.3 mesh-easymesh.0.9.5
```
```
File "easymesh/dune", line 17, characters 0-111:
17 | (rule
18 |  (targets EasyMesh.exe)
19 |  (deps    easymesh.c)
20 |  (action  (system "%{cc} -o EasyMesh.exe easymesh.c -lm")))
(cd _build/default/easymesh && C:\Windows\system32\cmd.exe /c "cl -nologo -O2 -Gy- -MD -o EasyMesh.exe easymesh.c -lm")
The command ""cl -nologo -O2 -Gy- -MD -o EasyMesh.exe easymesh.c -lm" is either misspelled or could not be found
```
No errors with
```bash
opam install dune.2.9.3 mesh-easymesh.0.9.5
```
or
```bash
opam install dune.1.11.4 mesh-easymesh.0.9.5
```

similar:
bin_prot.v0.12.0:
https://github.com/janestreet/bin_prot/blob/c4cfc29bfeadd69a644b6627f458ed66679aa37f/src/dune#L7

mpg123.0.3:
https://github.com/mbacarella/mpg123/blob/63c2d9be32892732d1957ad2df24d393fb747b2d/c/dune#L25

**Comment by rgrinberg** [comment]
@nojb @dra27 any idea as to what might be causing this?

**Comment by nojb** [comment]
No idea, but I am seeing correctly a repeated double quote in the error message?

![image](https://user-images.githubusercontent.com/113560/163685262-e32896e9-b0c7-4046-8fd4-817998015ba3.png)

**Comment by nojb** [comment]
I took a quick look, the difference is that up to 2.9 we used `Unix.create_process` to execute the rule action, see https://github.com/ocaml/dune/blob/dea03875affccc0620e902d28fed8d6b4351e112/src/stdune/spawn.ml#L34-L37 Starting in 3.0 we switched to the `spawn` library https://github.com/ocaml/dune/blob/9bf3da86f1de33a9884eb91610a6dd9c875ebcbe/vendor/spawn/src/spawn.ml#L123-L129 which calls `CreateProcess` directly, but with a slightly different quoting heuristics.

It turns out that the quoting heursitics of `Unix.create_process` are better than those in `spawn` library, causing the regression.

**Comment by nojb** [comment]
Fix in janestreet/spawn#29

**Comment by nojb** [comment]
@rgrinberg The vendored version of `spawn` needs to be updated for this to be *actually* fixed.
